---
title: "SARS-CoV-2 Variant Coinfection"
author: "A.M. Realingo"
date: "`r format(Sys.Date(), '%B %d, %Y')`"
output: pdf_document
---
## Section 1: Lineage assignment
  ```{r, echo=FALSE, warning=FALSE}
    library(knitr)
    library(magrittr)
    library(gt)
    library(kableExtra)
    data <- read.table(params$lineage_assignment, header = TRUE, sep = "\t")
    table_lineage <- kable(data)
    styled_table <- table_lineage %>%
    kable_styling(full_width = FALSE) %>%
    row_spec(0, bold = TRUE, color = "white", background = "#0073e6") %>%
    row_spec(c(1,3), bold = TRUE) %>%
    column_spec(1, bold = TRUE, color = "white", background = "#009933") %>%
    column_spec(2:3, italic = TRUE)

    styled_table
  ```

## Section 2: Freyja lineage abundance

  ![Freyja plot showing relative SARS-CoV-2 lineage abundance in the samples](`r params$freyja_plot`)

## Section 3: Virstrain lineage assignment for samples flagged for SARS-CoV-2 coinfection
  include_graphics(params$virstrain_html)

  virstrain_txt <- read.table(params$virstrain_txt, header = TRUE)
  kable(virstrain_txt)

## Section 4: Bammix flags for nucleotide mixture

  ![Bammix plot for nucleotide mixtures of flagged sample](`r params$bammix_plot`)

## Section 5: Alternative allele fraction per mutation

  ![Alternative allele fraction per mutation of sample flagged for SARS-CoV-2 coinfection](`r params$aafplot_mut`)

## Section 6: Alternative allele fraction per amplicon

  ![Alternative allele fraction per amplicon](`r params$aafplot_amp`)